octobot/community/authentication.py

```diff
@@ -201,19 +204,102 @@ async def login(self, email, password):
             raise authentication.FailedAuthentication(e)
         if self.is_logged_in():
             await self.update_supports()
-            await self._update_feed_device_uuid()
+            await self.update_selected_device()
+
+    async def update_selected_device(self):
+        self.user_account.flush_device_details()
+        await self._load_device_if_selected()
+        if not self.user_account.has_selected_device_data():
+            self.logger.info("No selected device. Please select a device to enable your community features.")
+
+    async def _load_device_if_selected(self):
+        # 1. use user selected device id if any
+        if saved_uuid := self._get_saved_gql_device_id():
+            await self.select_device(saved_uuid)
+        else:
+            # 2. fetch all user devices and create one if none, otherwise if there is only one, use it
+            await self.load_user_devices()
+            if len(self.user_account.get_all_user_devices_raw_data()) == 0:
+                await self.select_device(
+                    self.user_account.get_device_id(
+                        self.create_new_device()
+                    )
+                )
+            # 3. fetch all user devices and if there is only one, use it
+            if len(self.user_account.get_all_user_devices_raw_data()) == 1:
+                await self.select_device(
+                    self.user_account.get_device_id(
+                        self.user_account.get_all_user_devices_raw_data()[0]
+                    )
+                )
```

Comment:
Are we sure we want to do that? I a user has already a device it could be because it has an OctoBot running somewhere else, we shouldn't select the first device if a user already have one in my opinion.

Reply:
right, i'll remove this